### DISCUSSION TITLE
Error: Invalid signature from encrypted assertion

### DISCUSSION THREAD

**Discussion by awxlfy** [bergie/passport-saml#651] [Q&A]
Hi there! I had another issue stemming from the [key encryption algorithm](https://github.com/node-saml/passport-saml/issues/618) not being supported a little while back which has now been solved by my IdP. Now it seems that after a successful login I am redirected back correctly but presented with this error:

```
Error: Invalid signature from encrypted assertion
    at SAML.validatePostResponseAsync (/var/www/shibb/node_modules/passport-saml/lib/node-saml/saml.js:598:27)
```
    
I believe this is related to my cert and key (correct me if I'm wrong please) but I'm not 100% sure so wanted to find out.

Thanks!

**Comment by cjbarth** [comment]
There isn't much to go on here, so this isn't really a bug. I'll make this a discussion so more dialog can be had there.

**Reply by jpae316** [reply]
I am getting the same respond.
How was this resolved? Where is the discussion?

**Reply by aw1875** [reply]
It wasn't resolved unfortunately. I reached out yo my IdP but haven't had any luck with getting this working still.

**Comment by cjbarth** [comment]
Can you validate the signature of the encrypted assertion over at [samltool.com](https://www.samltool.com/online_tools.php)?

**Reply by aw1875** [reply]
I actually resolved this about a month ago. I forget exactly what the issue was but I believe a combo of problems on both my end and my IdP's end were solved and everything works properly now.
